edge-modules/TwinTester/TwinEdgeOperationsResultHandler.cs

```diff
@@ -43,27 +47,22 @@ public Task HandleTwinValidationStatusAsync(string status)
 
         public Task HandleReportedPropertyUpdateExceptionAsync(string propertyKey, string failureStatus)
         {
-            return this.SendReportAsync(StatusCode.ReportedPropertyUpdateCallFailure, propertyKey, failureStatus);
+            return Task.CompletedTask;
         }
 
-        async Task SendReportAsync(StatusCode statusCode, string details, string exception = "")
+        TwinCollection GetTwinCollection(string propertyKey)
         {
-            var result = new TwinTestResult() { Operation = statusCode.ToString(), TwinProperty = details, ErrorMessage = exception };
-            await ModuleUtil.ReportStatus(trcClient, Logger, this.moduleId, result.ToString(), TestOperationResultType.Twin.ToString());
+            var properties = new TwinCollection();
+            properties[propertyKey] = propertyKey;
+
+            return properties;
         }
 
-        class TwinTestResult
+        async Task SendReportAsync(string source, StatusCode statusCode, TwinCollection details, string exception = "")
         {
-            public string Operation { get; set; }
-
-            public string TwinProperty { get; set; }
-
-            public string ErrorMessage { get; set; }
-
-            public override string ToString()
-            {
-                return JsonConvert.SerializeObject(this, Formatting.Indented);
-            }
+            var result = new TwinTestResult() { Operation = statusCode.ToString(), Properties = details, ErrorMessage = exception, TrackingId = this.trackingId.GetOrElse(string.Empty) };
```

Comment:
` TrackingId = this.trackingId.GetOrElse(string.Empty) ` 
Are you sure you want to use an `string.Empty` instead of throwing an assert when `trackingId` is not set? 
I'd suggest using `TrackingId = this.trackingId.Expect(() => new ArgumentException("trackingId is null or empty"));`